### PR TITLE
Support for <indexterm>s when reading DocBook

### LINKED ISSUE #7427: <indexterm> support in DocBook reader?
The documentation within the DocBook reader [suggests](https://github.com/jgm/pandoc/blob/c8ab8bccf25ad01b9a98b2e59c84c151c5b0c371/src/Text/Pandoc/Readers/DocBook.hs#L211) that `<indexterm>` is supported. However, when I have `<indexterm>` elements, they are interpreted as just any regular old phrase.

Here's an example of 2 index terms in a DocBook source fragment:

```
<primary>IM</primary>
</indexterm>
<indexterm>
<primary>Instant Messaging</primary>
</indexterm>
```

These turn up in the JSON AST without anything to hook on for a filter:

```
{"t":"Str","c":"IM"},{"t":"SoftBreak"},{"t":"Str","c":"Instant"},{"t":"Space"},{"t":"Str","c":"Messaging"}
```

I'm using Pandoc to convert DocBook into LaTeX, and it works surprisingly well (except for a few Lua filters I had to write). `<indexterm>` support is, for now, the one missing piece. If you're interested to have this supported and not interested in implementing this (this being open source and all and you being free to spend your time as you please), I'll consider submitting a patch request.

### PR DESCRIPTION
**Pull request by bigsmoke**
This PR resolves #7427.  This was my first Haskell patch ever, so I'm sure that my additions, although functional, are an eyesore to seasoned Haskell coders. Please feel even more free than you usually would to correct/criticize my code.

### CODE REVIEW

File: src/Text/Pandoc/Readers/DocBook.hs [hunk 1]
```diff
@@ -1079,6 +1079,17 @@ elementToStr :: Content -> Content
 elementToStr (Elem e') = Text $ CData CDataText (strContentRecursive e') Nothing
 elementToStr x = x
 
+indextermNaryTextAsAttr :: Text -> Element -> Maybe (Text, Text)
+indextermNaryTextAsAttr n e = case findChild q e of
+        Nothing -> Nothing
+        Just naryEl -> Just (n, (strContent naryEl))
```
Comment (jgm) [inline src/Text/Pandoc/Readers/DocBook.hs:1085]: Here it's idiomatic to omit the parentheses around `(strContent naryEl)`.

**Comment by jgm** [comment]
Should we include the pagenum attribute?

What about `see` and `seealso` elements (there can be multiple of the latter)?

**Comment by jgm** [comment]
A test would be nice, too!
Easiest would be a "command test", since there isn't currently a Tests.Reader.DocBook module.

**Comment by bigsmoke** [comment]
> Should we include the pagenum attribute?

I don't think so. That is only used within `<index>` sections, which, like we discussed in #7608, makes most sense to continue to ignore in Pandoc.
 
> What about `see` and `seealso` elements (there can be multiple of the latter)?

I will follow up with a new PR request with (partial) support for `<see>` and `<seealso>` elements. They will be more straight-forward and less flaky to express as attributes to the span than `<primar>`, `<secondary>` and `<tertiary>`, because the latter actually have a mixed content model (which AsciiDoc doesn't have, so that it doesn't become a problem for me personally)..

> A test would be nice, too!
> Easiest would be a "command test", since there isn't currently a Tests.Reader.DocBook module.

I will included a test too. Thank you for the pointer! ☺

**Comment by bigsmoke** [comment]
I have improved the patch. You stimulated me to go a bit further and also do the Right Thing™ with mixed-content child element of `<indexterm>`, including a test case to ensure that the existing `strContentRecursive` that I reused indeed did what I hoped it would: to flatten the mixed content of, for example, a `<primary>` element into a string suitable for use in an attribute.

Thanks for sticking with me, @jgm ☺

**Comment by bigsmoke** [comment]
I reworded one of my commit messages and did a `git push --force-with-lease`, because one of the GitHub sanity check choked on its length 5 days ago.

**Comment by jgm** [comment]
Great, thanks.

**Pull request merged**